drivers/mtd/nvblk.c

```diff
@@ -0,0 +1,599 @@
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct block_operations g_nvblk_bops =
+{
+  nvblk_open,     /* open     */
+  nvblk_close,    /* close    */
+  nvblk_read,     /* read     */
+  nvblk_write,    /* write    */
+  nvblk_geometry, /* geometry */
+  nvblk_ioctl     /* ioctl    */
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  , nvblk_unlink  /* unlink   */
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int nvblk_convert_result(int err)
+{
+  switch (err)
+    {
+      case 0:
+        return 0;
+      case -NVB_ENOENT:
+        return -ENOENT;
+      case -NVB_EINVAL:
+        return -EINVAL;
+      case -NVB_EROFS:
+        return -EROFS;
+      case -NVB_EAGAIN:
+        return -EAGAIN;
+      case -NVB_ENOSPC:
+        return -ENOSPC;
+      default:
+        return -EFAULT;
+    }
+}
+
+/****************************************************************************
+ * Name: nvblk_open
+ *
+ * Description: Open the block device
+ *
+ ****************************************************************************/
+
+static int nvblk_open(FAR struct inode *inode)
+{
+  FAR nvblk_dev_t *dev;
+
+  DEBUGASSERT(inode->i_private);
+  dev = inode->i_private;
+  nxmutex_lock(&dev->lock);
+  dev->refs++;
+  nxmutex_unlock(&dev->lock);
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: nvblk_close
+ *
+ * Description: close the block device
+ *
+ ****************************************************************************/
+
+static int nvblk_close(FAR struct inode *inode)
+{
+  FAR nvblk_dev_t *dev;
+
+  DEBUGASSERT(inode->i_private);
+  dev = inode->i_private;
+  nxmutex_lock(&dev->lock);
+  dev->refs--;
+  nxmutex_unlock(&dev->lock);
+
+  if (dev->refs == 0 && dev->unlinked)
+    {
+      nxmutex_destroy(&dev->lock);
+      kmm_free(dev->pagebuf);
+      kmm_free(dev);
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Name: nvblk_read
+ *
+ * Description:  Read the specified number of sectors
+ *
+ ****************************************************************************/
+
+static ssize_t nvblk_read(FAR struct inode *inode,
+                          FAR unsigned char *buffer,
+                          blkcnt_t start_sector,
+                          unsigned int nsectors)
+{
+  FAR nvblk_dev_t *dev;
+  uint16_t bstart;
+  uint16_t bcnt;
+  int ret = 0;
+
+  DEBUGASSERT(inode->i_private);
+  dev = inode->i_private;
+  nxmutex_lock(&dev->lock);
+  bstart = start_sector << dev->log2_bpiob;
+  bcnt = nsectors << dev->log2_bpiob;
+
+  for (int n = 0; n < bcnt; n++)
```

Comment:
move n definition to begin of function and change to uint16_t

Reply:
Done